### PR TITLE
Feature/ws2 2953 unset graphql

### PR DESCRIPTION
**Pull request by JulissaDantes**
## Description
The users will now be able to call the `mutation Hide` special mutation to mark a stream as shouldIndex = false by passing the stream id.

### CODE REVIEW

File: packages/runtime/src/schema.ts [hunk 1]
```diff
@@ -178,6 +178,27 @@ const UpdateOptionsInput = new GraphQLInputObjectType({
   },
 })
 
+const CeramicSignerInputType = new GraphQLInputObjectType({
+  name: 'CeramicSignerInput',
+  fields: {
+    isAuthenticated: { type: GraphQLBoolean },
```
Comment (PaulLeCam) [inline packages/runtime/src/schema.ts:184]: What is the purpose of this input please?
Checking if the Ceramic instance has an authenticated DID should be handled internally by the context, not provided as input, if that's the goal here?

File: packages/runtime/src/schema.ts [hunk 2]
```diff
@@ -178,6 +178,27 @@ const UpdateOptionsInput = new GraphQLInputObjectType({
...
+  name: 'HideOptionsInput',
+  fields: {
+    asDID: {
+      type: GraphQLDID,
+      description: 'DID for the signer(Optional)',
+    },
+    signer: {
+      type: CeramicSignerInputType,
+      description: 'Ceramic signer (Optional)',
+    },
```
Comment (PaulLeCam) [inline packages/runtime/src/schema.ts:198]: Same as previous comment, this seems to be information we get from the context, not an input

**Review by PaulLeCam — CHANGES_REQUESTED** [review]

File: packages/runtime/src/schema.ts [hunk 3]
```diff
@@ -178,23 +178,12 @@ const UpdateOptionsInput = new GraphQLInputObjectType({
...
   name: 'HideOptionsInput',
   fields: {
-    asDID: {
-      type: GraphQLDID,
-      description: 'DID for the signer(Optional)',
-    },
-    signer: {
-      type: CeramicSignerInputType,
-      description: 'Ceramic signer (Optional)',
+    shouldIndex: {
```
Comment (PaulLeCam) [inline packages/runtime/src/schema.ts:209]: Could you use https://github.com/ceramicstudio/js-composedb/blob/8cdf4ad26012cc10cf12d757c801d8000d35558f/packages/runtime/src/schema.ts#L152 directly so we have a single reference for this field please?

File: packages/runtime/src/schema.ts [hunk 4]
```diff
@@ -178,23 +178,12 @@ const UpdateOptionsInput = new GraphQLInputObjectType({
...
 })
 
-const CeramicSignerInputType = new GraphQLInputObjectType({
-  name: 'CeramicSignerInput',
-  fields: {
-    isAuthenticated: { type: GraphQLBoolean },
-  },
-})
-
 const HideOptionsInput = new GraphQLInputObjectType({
```
Comment (PaulLeCam) [inline packages/runtime/src/schema.ts:202]: Could you rename the input as we discussed please?
```suggestion
const EnableIndexingOptionsInput = new GraphQLInputObjectType({
```

**Review by PaulLeCam — CHANGES_REQUESTED** [review]

File: packages/runtime/src/schema.ts (hunk 4 above)
Comment (JulissaDantes) [reply packages/runtime/src/schema.ts]: I had forgotten this, thanks for the reminder. Ready!

File: packages/runtime/src/context.ts [hunk 5]
```diff
@@ -71,6 +71,10 @@ export type Context = {
     content: Content,
     options?: UpsertOptions,
   ) => Promise<ModelInstanceDocument<Content> | null>
+  /**
+   * Enable indexing for an existing document.
+   */
+  enableDocIndexing: (model: string, opts?: UpdateOpts) => Promise<void>
```
Comment (PaulLeCam) [inline packages/runtime/src/context.ts:77]: I think it's confusing to name the stream ID `model` here as it's the stream that gets updated, nothing to do with the model

```suggestion
  enableDocIndexing: (id: string, opts?: UpdateOpts) => Promise<void>
```

File: packages/runtime/src/context.ts [hunk 6]
```diff
@@ -136,6 +140,18 @@ export function createContext(params: ContextParams): Context {
...
+      model: string,
+      opts?: UpdateOpts,
+    ): Promise<void> => {
+      const controller = getViewerID()
+      if (controller == null) {
+        throw new Error('Document can only be hidden with an authenticated account')
+      }
+
+      const doc = await loader.load<Content>({ id: model })
+      await doc.shouldIndex(false, opts)
```
Comment (PaulLeCam) [inline packages/runtime/src/context.ts:153]: The `shouldIndex` argument should be provided here so indexing can be disabled but also enabled back
```suggestion
      id: string,
      shouldIndex: boolean,
      opts?: UpdateOpts,
    ): Promise<void> => {
      const controller = getViewerID()
      if (controller == null) {
        throw new Error('Document can only be hidden with an authenticated account')
      }

      const doc = await loader.load<Content>({ id })
      await doc.shouldIndex(shouldIndex, opts)
```

File: packages/runtime/src/schema.ts [hunk 7]
```diff
@@ -1363,6 +1374,24 @@ class SchemaBuilder {
         return { document: await ctx.loader.update(input.id, input.content, input.options) }
       },
     })
+
+    this.#mutations[`enableIndexing${name}`] = mutationWithClientMutationId({
+      name: `EnableIndexing${name}`,
+      inputFields: () => ({
+        id: { type: new GraphQLNonNull(GraphQLID) },
+        options: { type: EnableIndexingOptionsInput },
```
Comment (PaulLeCam) [inline packages/runtime/src/schema.ts:1382]:
```suggestion
        shouldIndex { type: new GraphQLNonNull(GraphQLBoolean) },
```

File: packages/runtime/src/schema.ts [hunk 8]
```diff
@@ -1363,6 +1374,24 @@ class SchemaBuilder {
...
+      }),
+      outputFields: () => ({
+        ...queryFields,
+        document: { type: this.#types[name] },
+      }),
+      mutateAndGetPayload: async (input: { id: string; options?: UpdateOpts }, ctx: Context) => {
```
Comment (PaulLeCam) [inline packages/runtime/src/schema.ts:1388]:
```suggestion
      mutateAndGetPayload: async (input: { id: string; shouldIndex: boolean }, ctx: Context) => {
```

File: packages/runtime/src/schema.ts [hunk 9]
```diff
@@ -1363,6 +1374,24 @@ class SchemaBuilder {
...
+        if (ctx.ceramic.did == null || !ctx.ceramic.did.authenticated) {
+          throw new Error('Ceramic instance is not authenticated')
+        }
+        return { document: await ctx.enableDocIndexing(input.id, input.options) }
```
Comment (PaulLeCam) [inline packages/runtime/src/schema.ts:1392]:
```suggestion
        return { document: await ctx.enableDocIndexing(input.id, input.shouldIndex) }
```

File: packages/runtime/src/schema.ts [hunk 10]
```diff
@@ -199,6 +199,17 @@ const UpdateOptionsInput = new GraphQLInputObjectType({
...
+  name: 'EnableIndexingOptionsInput',
+  fields: {
+    streamID: {
+      type: GraphQLString,
+      description: 'Reference a specific and unique event stream',
+    },
+    shouldIndexField,
+  },
+})
+
```
Comment (PaulLeCam) [inline packages/runtime/src/schema.ts:212]: I don't think this is needed, the top-level fields of the mutation input should provide all the arguments
```suggestion
```

**Review by PaulLeCam — CHANGES_REQUESTED** [review]
Thanks for the changes, I just noted a few things with using the term "model" for the doc ID and things that could be simplified I think

**Review by PaulLeCam — APPROVED** [review]
Thanks for the changes!

**Pull request merged**